app/src/main/java/ch/epfl/sdp/appart/scrolling/ScrollingViewModel.java

```diff
@@ -32,9 +33,17 @@ public ScrollingViewModel(DatabaseService database) {
      * Gets the cards from the database and updates the LiveData list
      */
     public void initHome() {
-
         CompletableFuture<List<Card>> queriedCards = db.getCards();
         queriedCards.thenAccept(mCards::setValue);
+    }
+
+    public void filter(String location){
+        if(location != null && !location.equals("")) {
+            CompletableFuture<List<Card>> queriedCards = db.getCardsFilter(location);
+            queriedCards.thenAccept(mCardsFilter::setValue);
```

Comment:
I suggest calling `initHome` if the future fails, or at least show a toast.

Reply:
Yes, in case of errors in the database an other  attempt to recover all the data with initHome could be a better idea.